### DISCUSSION TITLE
How do remove power on servo between movements.

### DISCUSSION THREAD

**Discussion by Eebel** [ArminJo/ServoEasing#44] [Q&A] [answered]
Hi, I have this Arduino Nano sketch that is intended to run a "Bad Motivator" sequence for my Star Wars R5-D4 droid.  There are four buttons that I am using to test and tune the sequence on my workbench.  However, the sequence will be triggered from a remote when installed in my droid.  Everything works as I expect.  However, I have an issue with the servos always having power.  In this application, I always want to move the servo and then remove power from the servo so I don't waste battery power and also so the servo does not heat up and burn out.  Since the droid can be powered for 8-10 hours at a time, and the Bad Motivator sequence will only be used 10-12 times an hour, I am worried I will burn out the servo.  

I have tried using the .detach() command.  However, after that command no further commands are received by the servo.  I have tried reattaching the servo before each use.  However, that also causes issues in unintended movements.  I feel like I am close, but missing something obvious.  Can anyone help?

Tim

`void loop() {
  //Check button states
  buttonReset.loop();
  buttonTestPump.loop();
  buttonTestSmoke.loop();
  buttonTestAll.loop();

  if(buttonReset.isPressed()){
    //Servo1.setEasingType(EASE_BOUNCE_IN); 
    Servo1.easeTo(servoClosed,45); //close at 45/sec
    updatePixels = false; //stop updates to pixels
    ring1.ColorWipe(0,0,0);  //turn NeoPixel ring off;    
    #ifdef DEBUG
      Serial.println("Button Reset Pushed");
    #endif
  }
  //Run pump only-toggle
  //Also run servo limits
  if(buttonTestPump.isPressed()){
     //Toggle the Button state      
        statePumpButton = (!statePumpButton);
       digitalWrite(PIN_PUMPRELAY, statePumpButton);
       if (statePumpButton){
         //Open Servo
         //Servo1.setEasingType(EASE_BOUNCE_OUT);    
         Servo1.easeTo(servoOpen,360);//was 360
        #ifdef DEBUG
          Serial.println("buttonTestPump Pushed to Open Servo");
          Serial.print(" statePumpButton: ");
          Serial.println (statePumpButton);
          Serial.print(" servoOpen Value: ");
          Serial.println (servoOpen);          
        #endif         
        }          
        else{
          //Close Servo 
          //Servo1.setEasingType(EASE_BOUNCE_IN); 
          Servo1.easeTo(servoClosed,45);
          #ifdef DEBUG
            Serial.println("buttonTestPump Pushed to close Servo");
            Serial.print("statePumpButton: ");
            Serial.println(statePumpButton);
            Serial.print(" servoClosed Value: ");
            Serial.println (servoClosed);             
          #endif                     
        }      
  }
  //run Smoke..but turnn pump on
  if(buttonTestSmoke.isPressed()){
    stateSmokeButton = !stateSmokeButton;
    digitalWrite(PIN_PUMPRELAY, stateSmokeButton);//turn pump ON
    digitalWrite(PIN_SMOKERELAY, stateSmokeButton);
    #ifdef DEBUG
      Serial.println("buttonTestSmoke Pushed");
      Serial.print("stateSmokeButton: ");
      Serial.println(stateSmokeButton);
    #endif
  }
  //Run Entire sequence
  if(buttonTestAll.isPressed()){
    //stateTestAllButton = !stateTestAllButton;
    #ifdef DEBUG
      Serial.println("buttonTestAll Pushed");
      Serial.print("stateTestAllButton: ");
      Serial.println(stateTestAllButton);
    #endif
    isTriggered = true;
  }
 
 if(digitalRead(PIN_TRIGGER) == 0){
   //Trigger signal from user detected
  isTriggered = true;//digitalRead(PIN_TRIGGER);
      #ifdef DEBUG
        Serial.println("PIN_TRIGGER State Read as:");
        //Serial.print("stateTestAllButton: ");
        Serial.println(isTriggered);
      #endif

 }

  
  if(isTriggered && !wasTriggered){
    //start smoke sequence only if isTriggered is true and wasTriggered is false
    isTriggered = false;
    //set wasTriggered to true to keep it from firing twice without finishing.
    wasTriggered = true; //Set the sequence active flag to true
   //Store the times of the events for easy reference
    smokeOnMillis = millis() + 1000;
    smokeOffMillis = millis() + 4000;
    pumpOffMillis = millis() + 5000;
    //Turn airpump ON
    digitalWrite(PIN_PUMPRELAY, HIGH);  
    Servo1.easeTo(servoOpen,360);
    
    #ifdef DEBUG
    Serial.println("isTriggered was true;");
    #endif
  }
  if (wasTriggered){
    updatePixels = true;
    //ring1.update();
    if(millis()>pumpOffMillis){ //should run this last
      digitalWrite(PIN_PUMPRELAY, LOW);  //Pump OFF and reset flags for a subsequent trigger
      isTriggered = false; //set this false again incase we got a second command while running.
      wasTriggered = false;
      //ring1.ColorWipe(0,0,0);  //turn NeoPixel ring off;
    }
    else if(millis()> smokeOffMillis){//should run this second
      digitalWrite(PIN_SMOKERELAY,LOW); //Smoke OFF
    }
    else if(millis()> smokeOnMillis){//should run this first
      digitalWrite(PIN_SMOKERELAY, HIGH); //Smoke ON
      //Open Servo
         //Servo1.setEasingType(EASE_BOUNCE_OUT);  
         Servo1.easeTo(servoOpen,360);//was 360
    }
  }    
if(updatePixels){
  ring1.update();  
}  
    // Servo1.update();
    // if(!Servo1.isMoving()){
    //   Servo1.detach();
    // }
}`

**Comment by ArminJo** [comment]
Servos should not use much current after they reached the requested position, since they must be working in model aircrafts with small batteries.

**Reply by Eebel** [reply]
It's not about the power used as mush as running a servo constantly heats it if it is not perfectly aligned.  These droids are constantly moving over bumpy surfaces and his head literally spins.  The servos are not in a stable environment and when in an inactive state it is best to have them unpowered.

**Comment by ArminJo** [comment, marked as answer]
Ok, I got it.
After detach, the servo is not "powered" any more.
I added sample code at the end of the OneServo example.
